src/App.js

```diff
@@ -0,0 +1,164 @@
+import React, { useState } from 'react';
+function Calculator() {
+  const [result, setResult] = useState('');
+
+  const handleClick = (e) => {
+    setResult(result.concat(e.target.name));
+  };
+
+  const handleClear = () => {
+    setResult('');
+  };
+
+  const handleDelete = () => {
+    setResult(result.slice(0, -1));
+  };
+
+  const handleEqual = () => {
+    try {
+      setResult(eval(result).toString());
+    } catch (error) {
+      setResult('Error');
+    }
+  };
+
+  return (
+    <div className="flex flex-col items-center justify-center h-screen bg-gray-200">
+      <div className="w-80 bg-white rounded-lg shadow-lg p-4">
+        <input
+          type="text"
+          className="w-full bg-gray-100 rounded-lg p-2 mb-2"
+          value={result}
```

Comment:
There is no need to use the Input tag to display the result as the user won't be editing the result manually.